test/testFromStyleguidist.js

```diff
@@ -49,11 +49,12 @@ const testFromStyleguidist = (name, markdown, require) => {
         })
         done()
       }
-      codes.forEach(code => {
+      codes.forEach(async code => {
         const root = mount(
           <Preview code={code.content} evalInContext={evalInContext} />,
           options
         )
+        await sleep(100)
```

Comment:
Can you add a comment telling us why is this necessary ? I am a bit unconfortable to add this delay to all tests when it is necessary only for one. Is it not possible to specify the delay only for ActionMenu ?

Reply:
Sorry, I wasn't done yet! Since the test seems more flaky on the CI than on my machine, I first wanted to confirm that the sleep *really* works, including on the CI.

But yes, I wouldn't like it to stay that way either. I haven't found a way to *not* use a timer at all, but running it only for components that need it as you suggest is a good idea, thanks 👍

Reply:
Have you understood why the test is flaky, checking the code in material ui ?

Reply:
> Have you understood why the test is flaky, checking the code in material ui ?

The problem comes from popper-core, but no I haven't dug into the source to see what is going on. All I can say for sure is that the return value of `usePopper` changes multiples times, and it seems to come from popper-core and not react-popper itself.